geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java

```diff
@@ -1797,6 +1816,22 @@ public static String getSenderId(String regionName) {
     return regionName.substring(1, queueStringStart);
   }
 
+  public boolean isThereEventsMatching(Object object, Predicate condition) {
```

Comment:
If you know that `object` is always a `GatewaySenderEventImpl` then it would be better to use that in the method signature:
```
public boolean isThereEventsMatching(GatewaySenderEventImpl value, Predicate<Object> condition) {
```
...rather than casting it on the 1st line of the method.